src/main.rs

```diff
@@ -708,14 +709,17 @@ fn compile(
     entrypoint: Option<&str>,
     language: Option<&str>,
     settings: &HashMap<String, LanguageSettings>,
-    track_code_files: &mut HashSet<PathBuf>,
+    track_code_files: &mut HashMap<PathBuf, Option<PathBuf>>,
 ) -> Fallible {
     eprintln!("Compiling file {}", file_name.display());
 
     let mut entries = parser.get_entry_points(&document, language);
 
     let file_name_without_ext = file_name.with_extension("");
-    entries.insert(entrypoint, &file_name_without_ext);
+    entries.insert(
+        entrypoint,
+        (&file_name_without_ext, Some(PathBuf::from(file_name))),
```

Comment:
Wouldn't `file_name.to_owned()` do the same yet better capture the intent?

Reply:
Thanks! Was not aware of `Path::to_owned() -> PathBuf`